packages/rum-core/src/common/patching/history-patch.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * MIT License
+ *
+ * Copyright (c) 2017-present, Elasticsearch BV
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+import { INVOKE, HISTORY_PUSHSTATE } from '../constants'
+var alreadyPatched = false
+
+export function patchHistory(callback) {
```

Comment:
It's safer to do this patch after we have captured the page load event transaction, if the user registers a state in history using 'pushState', we will accidently capture all our requests as part of route change transaction instead of page-load which will be a broken behaviour..

Would like to have rigorous E2E tests for this change.

Reply:
I considered this to be a feature, that we will capture the first route change as a part of the page-load transaction. This will give the user a complete perspective to the initial page load.

Reply:
I am really not sure if this a good feature to be honest, It would simply increase the number of transactions count on the details page plus instead of focusing on the transactions that matter the most, we are simply polluting the UI more less useful `redirected transactions`. 

For the first phase at-least, we can simply drop them and add it if there is a huge interest from users.

Reply:
I'll investigate this scenario more

Reply:
@vigneshshanmugam , I will add e2e tests to #265 for this scenario

Reply:
Sure, sounds good 👍